src/hooks/usePagination.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo, useEffect, useState } from "react";
+
+/**
+ * A hook that handles pagination.
+ * @param data The data array to paginate.
+ * @param itemsPerPage Number of items per page. Returns all items if no value has been provided.
+ * @param initialPage?: Initial page number. Defaults to 1.
```

Comment:
Is `?:` the right syntax? Is the type hint enough to show that this is optional?